kymatio/scattering3d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,154 @@
+        ("standard","local","integral"). Default: "standard"
+    points: array-like, optional
+        List of locations in which to sample wavelet moduli. Used when
+        method == 'local'
+    integral_powers: array-like
+        List of exponents to the power of which moduli are raised before
+        integration. Used with method == 'standard', method == 'integral'
+    """
+    def __init__(self, J, shape, L=3, sigma_0=1, max_order=2, rotation_covariant=True, method='standard', points=None,
+                 integral_powers=(0.5, 1., 2.), backend='torch'):
+        ScatteringTorch.__init__(self)
+        ScatteringBase3D.__init__(self, J, shape, L, sigma_0, max_order, rotation_covariant, method, points,
+                 integral_powers, backend)
+        ScatteringBase3D._instantiate_backend(self, 'kymatio.scattering3d.backend.')
+        ScatteringBase3D.build(self)
+        ScatteringBase3D.create_filters(self)
+
+        self.register_filters()
+        self.J = J
+        self.shape = shape
+        self.L = L
+        self.sigma_0 = sigma_0
+
+        self.max_order = max_order
+        self.rotation_covariant = rotation_covariant
+        self.method = method
+        self.points = points
+        self.integral_powers = integral_powers
+        self.backend = backend
+
+        self.build()
+
+    def register_filters(self):
+        # transfer the filters from numpy to torch
+        for k in range(len(self.filters)):
+            filt = torch.zeros(self.filters[k].shape + (2,))
+            filt[..., 0] = torch.from_numpy(self.filters[k].real).reshape(self.filters[k].shape)
+            filt[..., 1] = torch.from_numpy(self.filters[k].imag).reshape(self.filters[k].shape)
+            self.filters[k] = filt
+            self.register_buffer('tensor' + str(k), self.filters[k])
+
+        g = torch.zeros(self.gaussian_filters.shape + (2,))
+        g[..., 0] = torch.from_numpy(self.gaussian_filters.real)
+        self.gaussian_filters = g
+        self.register_buffer('tensor_gaussian_filter', self.gaussian_filters)
+
+    def scattering(self, input_array):
+        buffer_dict = dict(self.named_buffers())
+        for k in range(len(self.filters)):
+            self.filters[k] = buffer_dict['tensor' + str(k)]
+
+        methods = ['standard', 'local', 'integral']
+        if (not self.method in methods):
+            raise (ValueError('method must be in {}'.format(methods)))
+        if self.method == 'integral': \
+                self.averaging = lambda x, j: self.backend.compute_integrals(self.backend.fft(x, inverse=True)[..., 0],
+                                                                             self.integral_powers)
+        elif self.method == 'local':
+            self.averaging = lambda x, j: \
+                self.backend._compute_local_scattering_coefs(x,
+                                                             self.tensor_gaussian_filter, j, self.points)
+        elif self.method == 'standard':
```

Comment:
this option might mean that we need to handle our outputs in a way that can keep space dimensions. relevant to the final reshapes

Reply:
This is fixed by counting forwards along the axes (not backwards).